### PR TITLE
Added day 2 operation for adding a bare metal node.

### PR DESCRIPTION
**Pull request by johnwilkins**
Signed-off-by: John Wilkins <jowilkin@redhat.com>
@rlopez133 
@iranzo 
# Description

Took draft from Roger and re-worked it a bit. The main changes are: 

I've added the steps: 
1. **Get a count of the worker nodes.** This is because if they already installed n number of worker nodes, and we don't know how many they have, we need that number to scale the nodes say from 2 to 3, because it appears that the replicas is the hard count, it's not incrementing them by the count.
2. **Get the machine set.** We don't know what the name of the machine set is. So I added that step so that it's obvious. 

I've consolidated the step: 

- **The baremetal node changes to provisioning status.** step with **When the node is provisioned it moves to provisioned status.** The command line usage is identical. What we're talking about is monitoring the state transition. 

I've also broken multiple commands in one block into multiple blocks, and put outputs from commands into their own blocks. I used the same corresponding steps that I had in my draft for cleaner punctuation and style compliance. 

Fixes https://issues.redhat.com/browse/OSDOCS-1308

I will have to create a separate downstream version with a separate assembly.

## Type of change

Please select the appropiate options:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist

- [x ] I have performed a self-review of my own code
- [ ] If a change is adding a feature, it should require a change to the README.md and the review should catch this.
- [ ] If the change is a fix, it should have an issue. The review should make sure the comments state the issue (not just the number) and it should use the keywords that will close the issue on merge.
- [ ] A change should not being merged unless it passes CI or there is a comment/update saying what testing was passed.
- [ ] PRs should not be merged unless positively reviewed.

### CODE REVIEW

File: documentation/ipi-install/modules/ipi-install-upstream-adding-a-worker-node-to-an-existing-cluster.adoc [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    address: ipmi://<bmc-ip>
+    credentialsName: openshift-worker-<num>-bmc-secret
+----
++
+Replace `<num>` for the worker number of bare metal node in two `name` fields and `credentialsName` field. Replace `<base64-of-uid>` with the `base64` string of the username. Replace `<base64-of-pwd>` with the `base64` string of the password. Replace `<NIC1-mac-address>` with the MAC address of the bare metal node's first NIC. Replace `<bmc-ip>` with the IP address of the bare metal node's baseboard management controller.
+
+[NOTE]
+====
+When using `redfish` or `redfish-virtualmedia`, add the
+appropriate addressing as described in the BMC addressing section. See xref:ipi-install-bmc-addressing_{context}[BMS addressing] for details.
```
Comment (rlopez133) [inline documentation/ipi-install/modules/ipi-install-upstream-adding-a-worker-node-to-an-existing-cluster.adoc:62]: spelled wrong, should be BMC not BMS, but it doesn't properly link

File: documentation/ipi-install/modules/ipi-install-upstream-adding-a-worker-node-to-an-existing-cluster.adoc [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+====
+
+. Create the bare metal node.
++
+[source,bash]
+----
+[kni@openshift-master-0 ~]$ oc -n openshift-machine-api create -f bmn.yaml
```
Comment (rlopez133) [inline documentation/ipi-install/modules/ipi-install-upstream-adding-a-worker-node-to-an-existing-cluster.adoc:69]: lets change this to from bmn.yaml to bmh.yaml to be mroe aligned with how its called 'baremetal host'

File: documentation/ipi-install/modules/ipi-install-upstream-adding-a-worker-node-to-an-existing-cluster.adoc [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+baremetalhost.metal3.io/openshift-worker-<num> created
+----
++
+Where `<num>` will be the worker number.
+
+. Power up and inspect the bare metal node.
++
+[source,bash]
+----
+[kni@openshift-master-0 ~]$ oc -n openshift-machine-api get bmn openshift-worker-<num>
```
Comment (rlopez133) [inline documentation/ipi-install/modules/ipi-install-upstream-adding-a-worker-node-to-an-existing-cluster.adoc:84]: command is `bmh` not bmn`

File: documentation/ipi-install/modules/ipi-install-upstream-adding-a-worker-node-to-an-existing-cluster.adoc [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+----
+NAME                 STATUS   PROVISIONING STATUS   CONSUMER   BMC                 HARDWARE PROFILE   ONLINE   ERROR
+openshift-worker-<num>   OK       ready                            ipmi://<out-of-band-ip>   unknown            true
+----
+
+. Ensure the `PROVISIONING STATUS` is `ready` before provisioning the bare metal node.
++
+[source,bash]
+----
+[kni@openshift-master-0 ~]$  oc -n openshift-machine-api get bmn openshift-worker-<num>
```
Comment (rlopez133) [inline documentation/ipi-install/modules/ipi-install-upstream-adding-a-worker-node-to-an-existing-cluster.adoc:99]: command is `bmh` not `bmn`

File: documentation/ipi-install/modules/ipi-install-upstream-adding-a-worker-node-to-an-existing-cluster.adoc (hunk 2 above)
Comment (rlopez133) [reply documentation/ipi-install/modules/ipi-install-upstream-adding-a-worker-node-to-an-existing-cluster.adoc:69]: everywhere we say openshift-master-0, I think we should label this provisioner. If we decide not to go that route and use master-0 then we need to export the KUBECONFIG to the master host as those commands won't work without that.

File: documentation/ipi-install/modules/ipi-install-upstream-adding-a-worker-node-to-an-existing-cluster.adoc [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+[source,bash]
+----
+[kni@openshift-master-0 ~]$ oc get nodes
+----
+
+. Get the machine set.
++
+[source,bash]
+----
+[kni@openshift-master-0 ~]$ oc get machinesets -n openshift-machine-api
```
Comment (rlopez133) [inline documentation/ipi-install/modules/ipi-install-upstream-adding-a-worker-node-to-an-existing-cluster.adoc:121]: we should show output.

File: documentation/ipi-install/modules/ipi-install-upstream-adding-a-worker-node-to-an-existing-cluster.adoc [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+You can also check the kubelet.
++
+[source,bash]
+----
+[kni@openshift-master-0 ~]$ ssh openshift-worker-<num>
+----
++
+[source,bash]
+----
+[kni@openshift-worker-<num>]$ journalctl -fu kubelet
```
Comment (rlopez133) [inline documentation/ipi-install/modules/ipi-install-upstream-adding-a-worker-node-to-an-existing-cluster.adoc:181]: after this, comes 13.3.1 which is in the wrong spot. 

Take a look https://deploy-preview-587--baremetal-deploy.netlify.app/4.6/deployment#ipi-install-upstream-dns-records-for-the-worker

File: documentation/ipi-install/modules/ipi-install-upstream-adding-a-worker-node-to-an-existing-cluster.adoc (hunk 1 above)
Comment (johnwilkins) [reply documentation/ipi-install/modules/ipi-install-upstream-adding-a-worker-node-to-an-existing-cluster.adoc]: We need to be using the [id="anchor_{context}"] approach were (context) is set in the master.adoc, for example: :context: some-context. We should not be using [[anchor-tag]].

File: documentation/ipi-install/modules/ipi-install-upstream-adding-a-worker-node-to-an-existing-cluster.adoc (hunk 6 above)
Comment (johnwilkins) [reply documentation/ipi-install/modules/ipi-install-upstream-adding-a-worker-node-to-an-existing-cluster.adoc:189]: What is the right spot?

File: documentation/ipi-install/modules/ipi-install-upstream-adding-a-worker-node-to-an-existing-cluster.adoc (hunk 1 above)
Comment (iranzo) [reply documentation/ipi-install/modules/ipi-install-upstream-adding-a-worker-node-to-an-existing-cluster.adoc]: It's been fixed with current 'master' version

File: documentation/ipi-install/modules/ipi-install-upstream-adding-a-worker-node-to-an-existing-cluster.adoc (hunk 6 above)
Comment (iranzo) [reply documentation/ipi-install/modules/ipi-install-upstream-adding-a-worker-node-to-an-existing-cluster.adoc:189]: @rlopez133 (ping)

**Pull request merged**